aws/hll/aws-s3-transfer-manager/src/download/worker.rs

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+use crate::download::context::DownloadContext;
+use crate::download::header;
+use crate::download::object_meta::ObjectMetadata;
+use crate::error;
+use crate::error::TransferError;
+use aws_sdk_s3::operation::get_object::builders::GetObjectInputBuilder;
+use aws_smithy_types::body::SdkBody;
+use aws_smithy_types::byte_stream::{AggregatedBytes, ByteStream};
+use std::ops::RangeInclusive;
+use std::{cmp, mem};
+use tokio::sync::mpsc;
+use tracing::Instrument;
+
+// FIXME - should probably be enum ChunkRequest { Range(..), Part(..) } or have an inner field like such
+#[derive(Debug, Clone)]
+pub(super) struct ChunkRequest {
+    // byte range to download
+    pub(super) range: RangeInclusive<u64>,
+    pub(super) input: GetObjectInputBuilder,
+    // sequence number
+    pub(super) seq: u64,
+}
+
+impl ChunkRequest {
+    /// Size of this chunk request in bytes
+    pub(super) fn size(&self) -> u64 {
+        self.range.end() - self.range.start() + 1
+    }
+}
+
+#[derive(Debug, Clone)]
+pub(crate) struct ChunkResponse {
+    // the seq number
+    pub(crate) seq: u64,
+    // chunk data
+    pub(crate) data: Option<AggregatedBytes>,
+    // object metadata
+    pub(crate) object_meta: Option<ObjectMetadata>,
```

Comment:
What does it mean for a `ChunkResponse` to have no data or metadata?

Reply:
I wondered that as well and started to change it but we can't actually construct `AggregatedBytes` directly as it turns out. Only other advantage is it makes testing a little easier to not have to set things like `ObjectMetadata`. 

I think this could probably change as we move away from `AggregatedBytes`